test/e2e/gardener/shoot/internal/inclusterclient/inclusterclient.go

```diff
@@ -53,8 +53,7 @@ func VerifyInClusterAccessToAPIServer(parentCtx context.Context, f *framework.Sh
 	ctx, cancel := context.WithTimeout(parentCtx, 2*time.Minute)
 	defer cancel()
 
-	objects := getObjects(f.Shoot.Spec.Kubernetes.Version)
-	prepareObjects(ctx, f.ShootClient.Client(), objects...)
+	defer prepareObjects(ctx, f.ShootClient.Client(), f.Shoot.Spec.Kubernetes.Version)()
```

Comment:
This somehow feels weird. It took me some time to understand that the `prepareObjects` call is actually executed as before and not deferred, but its return value (which is a function) is deferred.

I would prefer to have the immediate action not on the same line as the `defer` statement so that the normal assumption that everything on a `defer` line will be executed later. Something like below:

```suggestion
	cleanUpFunc := prepareObjects(ctx, f.ShootClient.Client(), f.Shoot.Spec.Kubernetes.Version)()
	defer cleanUpFunc()
```

Reply:
Fair enough, your suggestion is more readable. Done 👍